rust_src/src/fns.rs

```diff
@@ -90,6 +93,16 @@ pub fn quote(args: LispCons) -> LispObject {
     args.car()
 }
 
+#[lisp_fn]
+pub fn mapc(function: LispObject, sequence: LispObject) -> LispObject {
+    let leni = length(sequence) as EmacsInt;
```

Comment:
not sure if `leni` as name is ok
same for the cast but that's a simple solution for now

Reply:
Both are fine.